rest-api/vip-endpoints.php

```diff
@@ -65,11 +65,32 @@ public function rest_api_init() {
 	 */
 
 	/**
-	 * Some `/vip/` endpoints need to be accessible to requests from WordPress.com
+	 * Ensure `/vip/` endpoints are accessible to authenticated requests
+	 *
+	 * Circumvents any other authentication blocks and gives our method preference
 	 */
 	public function force_authorized_access( $result ) {
-		if ( 0 === strpos( $_SERVER['REQUEST_URI'], '/wp-json/vip/v1/sites' ) && wpcom_vip_go_rest_api_request_allowed( $this->namespace ) ) {
-			return true;
+		global $wp_rewrite;
+
+		if ( $wp_rewrite->using_permalinks() ) {
+			$rest_relative = get_rest_url( null, $this->namespace );
```

Comment:
This should either be trailing-slashed, or the checks below need to check that someone doesn't register `vip-fake` and have that work.